index.js

```diff
@@ -23,12 +23,16 @@ class NetworkResource extends Nanoresource {
     this._onclose = opts.close || noop
     this._onsocket = opts.socket || noop
     this._bootstrap = opts.bootstrap
-    this._ephemeral = opts.ephemeral !== false
-
+    this._ephemeral = opts.ephemeral
     this.utp.on('connection', this._onincoming.bind(this, false))
     this.tcp.on('connection', this._onincoming.bind(this, true))
   }
 
+  get ephemeral () {
+    if (!this.discovery) return this._ephemeral !== false
```

Comment:
return undefined or this._ephemeral directly here (or just don't have the getter, this is a low level module anyway)